relay-server/src/metrics_extraction/transactions.rs

```diff
@@ -316,22 +319,35 @@ fn extract_transaction_metrics_inner(
 
     // Measurements
     if let Some(measurements) = event.measurements.value() {
-        for (measurement_name, annotated) in measurements.iter() {
-            let measurement = match annotated.value().and_then(|m| m.value.value()) {
-                Some(measurement) => *measurement,
+        for (name, annotated) in measurements.iter() {
+            let measurement = match annotated.value() {
+                Some(m) => m,
+                None => continue,
+            };
+
+            let value = match measurement.value.value() {
+                Some(value) => *value,
                 None => continue,
             };
 
             let mut tags_for_measurement = tags.clone();
-            if let Some(rating) = get_measurement_rating(measurement_name, measurement) {
+            if let Some(rating) = get_measurement_rating(name, value) {
                 tags_for_measurement.insert("measurement_rating".to_owned(), rating);
             }
 
+            let stated_unit = measurement.unit.value().copied();
+            let default_unit = get_metric_measurement_unit(name);
+            if let (Some(default), Some(stated)) = (default_unit, stated_unit) {
+                if default != stated {
+                    relay_log::error!("unit mismatch on measurements.{}: {}", name, stated);
```

Comment:
I wonder if we can bring the SDK name into the error message, so we immediately know where this came from.

Reply:
The SDK is already on the context since we're in envelope processing. https://github.com/getsentry/relay/blob/2ca65bc87ce3c09274ffa2dd6858679649170f43/relay-server/src/actors/envelopes.rs#L1867-L1875


Generally, we should aim for central configuration rather than overloading code with error reporting boilerplate.